### PR TITLE
[IMS] API Update

### PR DESCRIPTION
**Pull request by Aloento**
https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/pull/2088

### CODE REVIEW

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
recheck

File: acceptance/openstack/ims/CreateWholeImageFromECS_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package ims
```
Comment (anton-sidelnikov) [inline acceptance/openstack/ims/CreateWholeImageFromECS_test.go]: lets call this file as usual with _, like whole_image_ecs_test.go

File: acceptance/openstack/ims/CreateImageFromECS_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package ims
```
Comment (anton-sidelnikov) [inline acceptance/openstack/ims/CreateImageFromECS_test.go]: image_ecs_test.go

File: acceptance/openstack/ims/CreateImageFromECS_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	fromDisk, err := images.CreateImageFromDisk(client2, images.CreateImageFromDiskOpts{
+		Name:      tools.RandomString("ims-test-", 3),
+		VolumeId:  ecs.VolumeAttached[0].ID,
+		OsVersion: "Debian GNU/Linux 10.0.0 64bit",
+	})
+	th.AssertNoErr(t, err)
+	jobEntities(t, client1, client2, fromDisk)
+
+	fromECS, err := images.CreateImageFromECS(client2, images.CreateImageFromECSOpts{
```
Comment (anton-sidelnikov) [inline acceptance/openstack/ims/CreateImageFromECS_test.go]: Also need to add: https://docs.otc.t-systems.com/image-management-service/api-ref/ims_apis/image/updating_image_information.html
and test for it

File: openstack/cbr/v3/policies/create.go [hunk 4]
```diff
@@ -29,7 +30,7 @@ type TriggerProperties struct {
 }
 
 func Create(client *golangsdk.ServiceClient, opts CreateOpts) (*Policy, error) {
-	b, err := golangsdk.BuildRequestBody(opts, "policy")
+	b, err := build.RequestBody(opts, "policy")
```
Comment (anton-sidelnikov) [inline openstack/cbr/v3/policies/create.go:33]: why cbr here?

File: internal/build/backward_test.go [hunk 5]
```diff
@@ -1,88 +0,0 @@
-package build
-
-import (
```
Comment (anton-sidelnikov) [inline internal/build/backward_test.go:3]: not related to this pr, revert please

File: openstack/compute/v2/servers/testing/requests_test.go [hunk 6]
```diff
@@ -171,6 +171,7 @@ func TestCreateServerWithUserdataEncoded(t *testing.T) {
 }
 
 func TestCreateServerWithImageNameAndFlavorName(t *testing.T) {
+	t.Skip("IMS endpoint changed")
```
Comment (anton-sidelnikov) [inline openstack/compute/v2/servers/testing/requests_test.go]: still needed?

File: acceptance/openstack/ims/CreateImageFromECS_test.go (hunk 2 above)
Comment (anton-sidelnikov) [reply acceptance/openstack/ims/CreateImageFromECS_test.go]: please use lower_case_and_underscore for file names

File: openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
+package images
+
+import (
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/build"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/common/tags"
+)
+
+// CreateWholeImageFromCBRorCSBSOpts Parameters in the request body when a CSBS backup or CBR backup is used to create a full-ECS image
+type CreateWholeImageFromCBRorCSBSOpts struct {
```
Comment (anton-sidelnikov) [inline openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go:10]: not supported in otc, please remove

File: openstack/ims/v2/members/GetMember.go [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+type GetMemberOpts struct {
+	ImageId  string `json:"-" required:"true"`
+	MemberId string `json:"-" required:"true"`
+}
+
+func GetMember(client *golangsdk.ServiceClient, opts GetMemberOpts) (*Member, error) {
```
Comment (anton-sidelnikov) [inline openstack/ims/v2/members/GetMember.go:15]: could you check if it works in otc?

File: openstack/ims/v2/tags/ListImagesTags.go [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package tags
```
Comment (anton-sidelnikov) [inline openstack/ims/v2/tags/ListImagesTags.go:1]: file duplicated

File: acceptance/openstack/ims/create_image_obs_test.go [hunk 10]
```diff
@@ -0,0 +1,62 @@
+package ims
```
Comment (anton-sidelnikov) [inline acceptance/openstack/ims/create_image_obs_test.go:1]: did you have tests for v1? I see here only v2

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: openstack/cbr/v3/policies/create.go (hunk 4 above)
Comment (Aloento) [reply openstack/cbr/v3/policies/create.go:33]: simple update(by the way)

File: internal/build/backward_test.go (hunk 5 above)
Comment (Aloento) [reply internal/build/backward_test.go:3]: related, as I sent you: an import cycle. It shouldn't be here.

File: openstack/compute/v2/servers/testing/requests_test.go (hunk 6 above)
Comment (Aloento) [reply openstack/compute/v2/servers/testing/requests_test.go]: I'd like to remove `TestCreateServerWithImageNameAndFlavorName`.

File: openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go (hunk 7 above)
Comment (Aloento) [reply openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go:10]: why not support?

File: openstack/ims/v2/members/GetMember.go (hunk 8 above)
Comment (Aloento) [reply openstack/ims/v2/members/GetMember.go:15]: it is just the same as the old `imageservice` (from OpenStack).
Test in the TF provider. (I don't have another ProjectId)
will try.

File: openstack/ims/v2/tags/ListImagesTags.go (hunk 9 above)
Comment (Aloento) [reply openstack/ims/v2/tags/ListImagesTags.go:1]: `ListImageTags` and `ListImagesTags` is diff.

File: acceptance/openstack/ims/create_image_obs_test.go (hunk 10 above)
Comment (Aloento) [reply acceptance/openstack/ims/create_image_obs_test.go:1]: Yes. It's already there.

File: acceptance/openstack/ims/CreateWholeImageFromECS_test.go (hunk 1 above)
Comment (Aloento) [reply acceptance/openstack/ims/CreateWholeImageFromECS_test.go]: done

File: acceptance/openstack/ims/CreateImageFromECS_test.go (hunk 2 above)
Comment (Aloento) [reply acceptance/openstack/ims/CreateImageFromECS_test.go]: done

File: openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go (hunk 7 above)
Comment (anton-sidelnikov) [reply openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go:10]: if it works then please add test

**Comment by anton-sidelnikov** [comment]
@Aloento in whole project we used lowercase for filenames, why are you using here CamelCase?

File: acceptance/openstack/ims/CreateImageFromECS_test.go (hunk 3 above)
Comment (Aloento) [reply acceptance/openstack/ims/CreateImageFromECS_test.go]: `openstack/ims/v2/images/UpdateImage.go`
already here with test.

File: openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go (hunk 7 above)
Comment (Aloento) [reply openstack/ims/v1/images/CreateWholeImageFromCBRorCSBS.go:10]: done

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by artem-lifshits — DISMISSED** [review]

File: openstack/imageservice/v2/images/types.go [hunk 11]
```diff
@@ -1,104 +0,0 @@
-package images
```
Comment (anton-sidelnikov) [inline openstack/imageservice/v2/images/types.go:1]: please revert that and use in structs

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
reboot

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
@anton-sidelnikov @artem-lifshits

**Pull request closed without merging**